### ISSUE TITLE
Several properties work in emulator but don't (or crash) on iPhone

### ISSUE THREAD

**Issue by cdelaet** [danielgindi/Charts#955]
When I set drawCircleEnabled = false on the dataset, my line chart looks correct (= without circles) on the emulator, but on an iPhone it still draws the circles. Same with lineWidth: the thick line only shows up on the emulator, while the iPhone displays the default thin line. 
Other properties such as fillColor even crash on the iPhone with EXC_BAD_ACCESS (but work on the emulator).
Strangely, setColor works fine both on the emulator and iPhone.
What am I doing wrong? 

Here's my test code (using autolayout):

    @IBOutlet var accelerationChart: LineChartView!

    var xVals: [Double] = [Double]()
    var yVals1: [ChartDataEntry] = [ChartDataEntry]()
    var xIndex = 0

     func showAccelerationChart() {
        for i in 0..<10 {
            xVals.append(Double(i))
            yVals1.append(ChartDataEntry(value: Double(i), xIndex: i))
        }
        
        let dataSet1: LineChartDataSet = LineChartDataSet(yVals: yVals1, label: "sec")
        dataSet1.drawCirclesEnabled = false
        // dataSet1.fillColor = UIColor.magentaColor() -- crashes on iPhone with EXC_BAD_ACCESS
        dataSet1.setColor(UIColor.magentaColor())
        dataSet1.lineWidth = 4.0

        var dataSets: [LineChartDataSet] = [LineChartDataSet]()
        dataSets.append(dataSet1)
        
        let data: LineChartData = LineChartData(xVals: xVals, dataSets: dataSets)
        
        accelerationChart.data = data
     }

**Comment by danielgindi** [comment]
You are probably using an old version on your iPhone while your Xcode for some reason does not know it needs to recompile.
Try to delete the app from the iPhone, then `Clean`, and `Build`.

**Comment by cdelaet** [comment]
I did as you told, but it sadly didn't help (removed app, clean, build, run again).
It's a new project and I had never used Charts before, so it was unlikely to be an old version which was stuck on the iPhone, I guess. I've just added Charts to my project a few days ago (using Carthage to add the framework to Embedded Binaries and dragging Charts.xcodeproj into my project, as per the Usage instructions). Cartfile.resolved shows I'm using v2.2.4
Would you have any other idea why it doesn't work for me? (Again, it works great on the emulator.)

**Comment by cdelaet** [comment]
I got it to work, Daniel, but I'm not very pleased with my solution. I removed the Carthage completely from my project and I'm now only using the Charts.xcodeproj file. I guess there was a conflict between the 2.
Do I still need Charts.xcodeproj when I'm already using Carthage? If I don't, the classes didn't show up in the Storyboard inspector, so I guess I do. I'm new at this, so hope you don't mind those questions.
